packages/eslint-plugin/src/rules/shorthand-property-sorting/__tests__/rule.test.ts

```diff
@@ -3,238 +3,757 @@ import { outdent } from 'outdent';
+        });
+
+        const Component = <div css={styles} />;
+        `,
+      errors: [
+        // Four violations, because
+        //
+        // * borderTop and border out of order
+        // * borderColor and borderTop out of order
+        //
+        // and two violations for EACH pair that is out of order.
+        { messageId: 'shorthand-first' },
+        { messageId: 'shorthand-first' },
+        { messageId: 'shorthand-first' },
+        { messageId: 'shorthand-first' },
+      ],
+    },
+
+    //
+    // includes pseudo-selectors -> pseudo and non-pseudo selectors is out of order
+    //
+
+    {
+      name: `includes pseudo-selectors -> pseudo and non-pseudo are out of order (css, ${imp})`,
+      code: outdent`
+        import { css } from '${imp}';
+        const styles = css({
+          '&:hover': {
+              borderTop: '1px solid #00b8d9', // 4
+              borderColor: 'red', // 2
+              border: '1px solid #00b8d9', // 1
+          },
+          borderTop: '1px solid #00b8d9', // 4
+          borderColor: 'red', // 2
+          border: '1px solid #00b8d9', // 1
+        });
+
+        const Component = <div css={styles} />;
+      `,
+      errors: [
+        { messageId: 'shorthand-first' },
+        { messageId: 'shorthand-first' },
+        { messageId: 'shorthand-first' },
+        { messageId: 'shorthand-first' },
+        { messageId: 'shorthand-first' },
+        { messageId: 'shorthand-first' },
+        { messageId: 'shorthand-first' },
+        { messageId: 'shorthand-first' },
+      ],
+    },
+
+    //
+    // works with cx
+    //
+
+    {
+      // https://atlassian.design/components/css/overview#cx
+      name: 'works with cx (css and JSX element)',
+      code: outdent`
+        import { cssMap, cx } from '@atlaskit/css';
+        const styles = cssMap({
+          root: { paddingTop: '...' },
+          success: {
+            padding: '...',
+          },
+        });
+
+        <div css={cx(styles.root, styles.success)} />
+      `,
+      errors: [
+        { messageId: 'shorthand-first' },
+        { messageId: 'shorthand-first' },
+        { messageId: 'shorthand-first' },
+        { messageId: 'shorthand-first' },
+      ],
+    },
+    {
+      // https://atlassian.design/components/css/overview#cx
+      name: 'works with cx (xcss and Primitives)',
+      code: outdent`
+        import { cssMap, cx } from '@atlaskit/css';
+        import { Box } from '@atlaskit/primitives';
+
+        const styles = cssMap({
+          root: { paddingTop: '...' },
+          success: {
+            padding: '...',
+          },
+        });
+
+        <Box xcss={cx(styles.root, styles.success)} />
+      `,
+      errors: [
+        { messageId: 'shorthand-first' },
+        { messageId: 'shorthand-first' },
+        { messageId: 'shorthand-first' },
+        { messageId: 'shorthand-first' },
+      ],
+    },
+
+    //
+    // false positives for cssMap
```

Comment:
do we care?

Reply:
This might just end up adding noise because while you _could_ statically resolve this if there's any valid scenario of them being mixed, it's a ton of work…

Would we basically just have to ignore every `cssMap` with this?

tl;dr: on if we care or not, `cssMap` might overtake `css` one day as we tell people to use `@atlaskit/css` for Primitives.

Reply:
fixed the false positives for when the cssMap key (or whatever we want to call it) can be easily determined at build time (i.e. it's a string)